### DISCUSSION TITLE
Custom getRepeatJobId?

### DISCUSSION THREAD

**Discussion by matiaslopezd** [taskforcesh/bullmq#1890] [Q&A]
Hi! I want to know if is possible to override the `getRepeatJobId` generation because Im trying to use `uuid` for repeatable and non-repeatable jobs, but only works for non-repeatable. (I read the docs [Jobs ID](https://docs.bullmq.io/guide/jobs/job-ids)) and [API reference](https://api.docs.bullmq.io/classes/Queue.html#add).

https://github.com/taskforcesh/bullmq/blob/0acfb79f779c6dcffca946f843290ae7bc4066e7/src/classes/repeat.ts#L204-L214

So the output of the `getRepeatJobId` could be `uuid` instead composition of `repeat:checksum:nextMilis`?

An finally, how works [id](https://api.docs.bullmq.io/classes/Job.html#id), [jobId](https://api.docs.bullmq.io/interfaces/BaseJobOptions.html#jobId), [jobId repeatable](https://api.docs.bullmq.io/interfaces/RepeatOptions.html#jobId) and [repeatJobKey](https://api.docs.bullmq.io/classes/Job.html#repeatJobKey)?

**Comment by manast** [comment]
We have a PR on the works that may do what you need: https://github.com/taskforcesh/bullmq/pull/1824
